### ISSUE TITLE
Does not identify syntax error on nested f-strings.

### ISSUE THREAD

**Issue by PotatoK123** [microsoft/pyright#5115]
Type: <b>Bug</b>

# Behaviour
## Expected vs. Actual

The Python extension reports no problems with the following code:
```py
print(f"{1+2:{1+2:{1+1:{1}}}}")
print(f'{123:{1:{2}}}')
```

The first line is a test case from https://github.com/python/cpython/blob/main/Lib/test/test_fstring.py#L568 and should report a syntax error (`SyntaxError: f-string: expressions nested too deeply`). The second line is my own test case and should also report the same syntax error. Other syntax errors are correctly identified by the extension however (and I can provide screenshots if needed).

## Steps to reproduce:

1. Copy and paste the code into a file.
2. Save the file.
3. No problems are reported.
4. Run and get a syntax error.
6. Comment first line.
7. Save file.
8. No problems are reported.
9. Run and get a syntax error.

# Diagnostic data

-   Python version (& distribution if applicable, e.g. Anaconda): 3.11.3
-   Type of virtual environment used (e.g. conda, venv, virtualenv, etc.): MicrosoftStore
-   Value of the `python.languageServer` setting: Default

<details>

<summary>Output for <code>Python</code> in the <code>Output</code> panel (<code>View</code>→<code>Output</code>, change the drop-down the upper-right of the <code>Output</code> panel to <code>Python</code>)
</summary>

<p>

Not sure what I was meant to put here. Maybe someone could clarify for me? The Output panel shows nothing by the way.

</p>
</details>

<details>

<summary>User Settings</summary>

<p>

```

languageServer: "Pylance"

```

</p>
</details>

Extension version: 2023.8.0
VS Code version: Code 1.78.2 (b3e4e68a0bc097f0ae7907b217c1119af9e03435, 2023-05-10T14:39:26.248Z)
OS version: Windows_NT x64 10.0.19045
Modes:
Sandboxed: Yes

Thanks for the work done on this extension! I've been using it for many years now. ❤️

**Comment by erictraut** [comment]
Looking into this in more detail, the limitation exists only for format string specifiers — the portion of the f-string after a colon.

I've added code to detect this case. It will be included in the next release of pyright.

**Comment by PotatoK123** [comment]
Thanks guys! Much appreciated! 👍 👍

**Comment by erictraut** [comment]
This is included in pyright 1.1.309, which I just published. It will also be included in a future release of pylance.
